src/rail/estimation/algos/deepdisc.py

```diff
@@ -167,44 +199,46 @@ def run(self):
         # optimizer = return_optimizer(cfg)
         optimizer = solver.build_optimizer(cfg_loader, model)
 
-        """
-        When using the single test dictionary, add this code and replace "mapper" below
-        
-        
+
+        #When using the single test dictionary, add this code and replace "mapper" below
         def dc2_key_mapper(dataset_dict):
             filename = dataset_dict["filename"]
             return filename
 
         IR = DC2ImageReader()
         mapper = RedshiftDictMapper(IR, dc2_key_mapper).map_data
-        """
+        
+
+        # mapper = RedshiftFlatDictMapper().map_data
 
-        mapper = RedshiftFlatDictMapper().map_data
         loader = d2data.build_detection_train_loader(
-            train_data, mapper=mapper, total_batch_size=batch_size
+            metadata, mapper=mapper, total_batch_size=batch_size
         )
-        test_loader = d2data.build_detection_test_loader(
-            train_data, mapper=mapper, batch_size=batch_size
+
+        eval_loader = d2data.build_detection_test_loader(
+            metadata, mapper=mapper, batch_size=batch_size
         )
 
         saveHook = return_savehook(output_name)
-        lossHook = return_evallosshook(val_per, model, test_loader)
+        lossHook = return_evallosshook(val_per, model, eval_loader)
         schedulerHook = return_schedulerhook(optimizer)
         hookList = [lossHook, schedulerHook, saveHook]
 
         trainer = return_lazy_trainer(
             model, loader, optimizer, cfg, cfg_loader, hookList
         )
 
+        # how often the trainer prints something to screen, could be a config
         trainer.set_period(5)
 
         print("Model training:")
         trainer.train(0, epochs)
 
+        #! Question for Grant, what is it that we actually want to save here?
         self.model = dict(nnmodel=model)
         self.add_data("model", self.model)
 
-
+#! I don't think we actually use this class???
 class DeepDiscEstimator(CatEstimator):
```

Comment:
Check with @grantmerz to determine if we actually use this class. If not, delete it, if so, remove this "I don't think we actually use this class" comment, and add a more substantial docstring that describes when this class should be used.

Reply:
Yeah, this was included for some initial tests I was doing, it can be removed

Reply:
I may not fully understand the goal here, but if the primary purpose of this class is to provide a point estimate of photo-z versus a pdf, and the point estimates are derived from the pdfs (i.e. mode, median, mean) then we can use the input parameter `calculated_point_estimates = ['mean', 'median', 'mode']` for the DeepDiscPDFEstimator to accomplish the same thing. 

The resulting point estimates will be packaged as ancillary data in the final qp.Ensemble output. 

If that meets your needs, then we can probably safely remove this class. But if not, then we should update it with the caching logic like what is present in DeepDiscPDFEstimator.